ee/webhooks/cmd/webhook-all-in-one.go

```diff
@@ -0,0 +1,186 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+
+	"github.com/formancehq/stack/libs/go-libs/auth"
+	"github.com/formancehq/stack/libs/go-libs/bun/bunconnect"
+	"github.com/formancehq/stack/libs/go-libs/licence"
+	"github.com/formancehq/stack/libs/go-libs/logging"
+	"github.com/formancehq/stack/libs/go-libs/otlp/otlptraces"
+	"github.com/formancehq/stack/libs/go-libs/publish"
+	"github.com/formancehq/stack/libs/go-libs/service"
+	"github.com/uptrace/bun"
+
+	"github.com/formancehq/webhooks/cmd/flag"
+
+	"github.com/formancehq/webhooks/internal/commons"
+	httpclient "github.com/formancehq/webhooks/internal/services/httpclient"
+
+	component "github.com/formancehq/webhooks/internal/components/commons"
+	webhookCollector "github.com/formancehq/webhooks/internal/components/webhook_collector"
+	whController "github.com/formancehq/webhooks/internal/components/webhook_controller"
+	webhookworker "github.com/formancehq/webhooks/internal/components/webhook_worker"
+
+	fxmodules "github.com/formancehq/webhooks/cmd/fx-modules"
+	httpserver "github.com/formancehq/webhooks/internal/services/httpserver"
+	storage "github.com/formancehq/webhooks/internal/services/storage/postgres"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"go.uber.org/fx"
+)
+
+func newAllInOneCommand() *cobra.Command {
+	return &cobra.Command{
+		Use:     "start",
+		Aliases: []string{"strt"},
+		Short:   "Run StandAlone Webhook",
+		RunE:    allInOneRun,
+		PreRunE: handleAutoMigrate,
+	}
+}
+
+func allInOneRun(cmd *cobra.Command, _ []string) error {
+
+	connectionOptions, err := bunconnect.ConnectionOptionsFromFlags(cmd.Context())
+	if err != nil {
+		return err
+	}
+
+	serviceInfo := commons.ServiceInfo{Name: ServiceName, Version: Version}
+	options := []fx.Option{
+		auth.CLIAuthModule(),
+		licence.CLIModule(ServiceName),
+		otlptraces.CLITracesModule(),
+		bunconnect.Module(*connectionOptions),
+		publish.CLIPublisherModule(ServiceName),
+		fx.Provide(
+
+			func() *http.Client {
+				return fxmodules.FxProvideHttpClient()
+			},
+			func(client *http.Client) *httpclient.DefaultHttpClient {
+				defaultClient := httpclient.NewDefaultHttpClient(client)
+				return &defaultClient
+			},
+
+			func() *component.RunnerParams {
+
+				runnerParams := flag.LoadRunnerParams()
+
+				return &runnerParams
+			},
+			func(auth auth.Auth, logger logging.Logger) httpserver.DefaultServerParams {
+
+				serverParams := httpserver.DefaultServerParams{}
+				serverParams.Addr = viper.GetString(flag.Listen)
+				serverParams.Auth = auth
+				serverParams.Info = serviceInfo
+				serverParams.Logger = logger
+
+				return serverParams
+			},
+
+			func(db *bun.DB) *storage.PostgresStore {
```

Comment:
In Go, you can use directly some things like `fx.Provide(storage.NewPostgresStoreProvider)` instead of wrapping the call.